File: email_test.ts
import { assertEquals, assertNotEquals } from "@std/assert";

import { normalizeEMAIL, validateEMAIL } from "./email.ts";

const varified_ids: string[] = [
  "a@localhost",
  "a.b@localhost",
  "b_2@example.edu",
  "b3@library.example.edu",
  "b4@agents.library.example.edu",
];

for (const id of varified_ids) {
  // Normalize
  const normalized = normalizeEMAIL(id);
  assertNotEquals(normalized, undefined);
  console.log(`Normalized Email: ${normalized}`);
  // Validate
  assertEquals(validateEMAIL(id), true);
}

File: email_test.go
package metadatatools

import "testing"

func TestEMAIL(t *testing.T) {
	verifiedIDs := []string{
		"a@localhost",
		"a.b@localhost",
		"b_2@example.edu",
		"b3@library.example.edu",
		"b4@agents.library.example.edu",
	}
	for _, id := range verifiedIDs {
		normalized := NormalizeEMAIL(id)
		if normalized == "" {
			t.Errorf("NormalizeEMAIL(%q) returned an empty string", id)
		}
		t.Logf("Normalized Email: %s", normalized)
		if !ValidateEMAIL(id) {
			t.Errorf("ValidateEMAIL(%q) = false, expected true", id)
		}
	}
}
